### PR TITLE
[T5A2][T09-B1]Bey Hao Yun

### PR DESCRIPTION
**Pull request by cardboardcode**
Likov's Substitution Principle states that if a program module is using a sub class, then
the reference to the super class can be replaced with a sub class without affecting the
functionality of the program module.
(Check web lecture for good example shown)

Making sure the sub class does not implement anything stricter than the parent class.

### CODE REVIEW

File: src/seedu/addressbook/commands/Command.java [hunk 1]
```diff
@@ -65,4 +65,5 @@ public int getTargetIndex() {
     public void setTargetIndex(int targetIndex) {
         this.targetIndex = targetIndex;
     }
+    public abstract boolean isMutating ();
```
Comment (cardboardcode) [inline src/seedu/addressbook/commands/Command.java]: You need to add a header here.

Comment (nirandiw) [inline src/seedu/addressbook/commands/Command.java]: Yes thats correct. It is better to mention it returns true if the Command *potentially* mutates the data (header comments specify a contract, so try to be as precise as possible).

File: src/seedu/addressbook/logic/Logic.java [hunk 2]
```diff
@@ -71,7 +71,10 @@ protected void setLastShownList(List<? extends ReadOnlyPerson> newList) {
     public CommandResult execute(String userCommandText) throws Exception {
         Command command = new Parser().parseCommand(userCommandText);
         CommandResult result = execute(command);
-        recordResult(result);
+
```
Comment (nirandiw) [inline src/seedu/addressbook/logic/Logic.java]: For the execute() method:
The behavior of the code has been changed. Ensure the header comment and the tests have been updated.

File: src/seedu/addressbook/logic/Logic.java [hunk 3]
```diff
@@ -71,7 +71,10 @@ protected void setLastShownList(List<? extends ReadOnlyPerson> newList) {
     public CommandResult execute(String userCommandText) throws Exception {
         Command command = new Parser().parseCommand(userCommandText);
         CommandResult result = execute(command);
-        recordResult(result);
+        
+        if (command.isMutating())
```
Comment (nirandiw) [inline src/seedu/addressbook/logic/Logic.java]: Coding standard violations!!!! Please refer the coding standard.

**Comment by nirandiw** [comment]
@cardboardcode Some comments added. Please close the PR after reading comments.

**Pull request closed without merging**